hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileReader.java

```diff
@@ -227,6 +261,40 @@ public BloomFilter readBloomFilter() {
     return records;
   }
 
+  public ClosableIterator<R> getRecordIterator(List<String> keys, Schema schema) throws IOException {
+    this.schema = schema;
+    reader.loadFileInfo();
+    Iterator<String> iterator = keys.iterator();
+    return new ClosableIterator<R>() {
+      private R next;
+      @Override
+      public void close() {
+        HoodieHFileReader.this.close();
+      }
```

Comment:
The tests fail with NPE i guess it is because this line, we should do nothing here because the caller would take care the close of the reader.

Reply:
![image](https://user-images.githubusercontent.com/36881649/156135938-9785a37e-03fc-493c-880a-218cac6ac1a0.png)
We change the return of lookupRecords to ClosableIterator, but `reader` is in the `try`, so `reader` was closed by the `lookupRecords`

Reply:
I think we should not close the file reader in the iterator, which make the reader can not be read anymore. The fetch of iterator should not have side effect except that we are sure the reader was only used for fetching the iterator then(still risky, we should only release the resource that belongs only to the iterator).

Reply:
yes, but the `reader` is newly created each time, and if it is not closed, it leaks.

Reply:
Can we keep the try block for the reader and do nothing in the iterator#close, also fix the checkstyle issue.